pyanaconda/modules/storage/partitioning/automatic/automatic_partitioning.py

```diff
@@ -191,6 +191,8 @@ def _check_reused_scheme(cls, storage, request):
             AUTOPART_TYPE_PLAIN: "partition",
         }
         for mountpoint in request.reused_mount_points:
+            if mountpoint in ["bootloader", "/boot/efi"]:
```

Comment:
I wonder what's the reason of skipping devices which are not relevant instead of just checking only for the relevant mount point. Aka
```
if mountpoint != '/':
    continue
```

We are checking that /home lives on the expected filesystem type, here, right?

Reply:
I think in the backend we should be rather restrictive to protect the data - in case we don't know prevent using the feature. In this case it means - we know that bootloader partitions don't have to has the type based on the partitioning scheme. For other mountpoints (for example separate /var) we don't know - so rather check and make it ignorelisted later than accept it and crash during the partitioning.  

So we are checking all reused mountpoints (except the ignorelisted) to be of expected type. Although the check will probably need to be refined in the future as use cases will grow.

Reply:
I think it could be worth adding the reasoning as a comment to the code in some sane form.